packages/core/src/controllers/TransactionsController.ts

```diff
@@ -124,9 +135,19 @@ export const TransactionsController = {
     })
   },
 
+  filterByConnectedChain(transactions: Transaction[]) {
+    const chainId = NetworkController.state.caipNetwork?.id
+    const filteredTransactions = transactions.filter(
+      transaction => transaction.metadata.chain === chainId
+    )
+
+    return filteredTransactions
+  },
```

Comment:
I thought we were requesting the filtered txs directly from blockchain api 🤔 

wouldn't this mess up with pagination?

Reply:
> wouldn't this mess up with pagination?

Mmm I'm not sure and I'm not able to test this 🤔

Reply:
@geekbrother any way we can do the filterin directly on blockchain API side? I think there's an edge case to this where if we filter txs and don't have enough to fill the ui, we will never request more txs even if you have them